relay-general/src/store/transactions/processor.rs

```diff
@@ -282,6 +282,11 @@ fn normalize_transaction_name(transaction: &mut Annotated<String>) -> Processing
             }
         }
 
+        if caps.is_empty() {
+            // Nothing to do for this transaction.
+            return Ok(());
+        }
```

Comment:
This also prevents copying the string from `trans` to `changed` and back if they are the same.